File: back-end/src/controllers/auth.controller.js
const authService = require('../services/authService');
const responseHelper = require('../utils/responseHelper');
const asyncHandler = require('../utils/asyncHandler');

const authController = {
  /**
   * Registra um novo usuário
   */
  register: asyncHandler(async (req, res) => {
    const user = await authService.register(req.body);
    return responseHelper.created(res, user, 'Usuário cadastrado com sucesso');
  }),

  /**
   * Autentica um usuário
   */
  login: asyncHandler(async (req, res) => {
    const { email, senha } = req.body;

    if (!email || !senha) {
      return responseHelper.error(res, 'Email e senha são obrigatórios', 400);
    }

    const user = await authService.login(email, senha);
    
    // Aqui você pode gerar um token JWT se quiser
    // Por enquanto, retornamos apenas os dados do usuário
    return responseHelper.success(res, {
      user,
      // token: 'seu-token-jwt-aqui' // Implementar JWT depois se necessário
    }, 'Login realizado com sucesso');
  }),
};

module.exports = authController;

File: backend/api/controllers/auth_controller.py
from rest_framework.decorators import api_view
from rest_framework.request import Request

from api.middleware.auth_middleware import get_user_from_authorization_header
from api.services import auth_service
from api.utils import response_helper


@api_view(["POST"])
def register(request: Request):
    try:
        result = auth_service.register_user(request.data)
        return response_helper.created(
            {"user": result["user"], "token": result["token"]},
            "Usuário cadastrado com sucesso",
        )
    except ValueError as exc:
        return response_helper.error(str(exc), status_code=400)


@api_view(["POST"])
def login(request: Request):
    email = request.data.get("email")
    senha = request.data.get("senha")

    if not email or not senha:
        return response_helper.error("Email e senha são obrigatórios", status_code=400)

    try:
        result = auth_service.login_user(email, senha)
        return response_helper.success(
            {"user": result["user"], "token": result["token"]},
            "Login realizado com sucesso",
        )
    except ValueError as exc:
        return response_helper.error(str(exc), status_code=400)


@api_view(["GET"])
def verify_token(request: Request):
    decoded = get_user_from_authorization_header(request.headers.get("Authorization"))
    return response_helper.success(
        {"user": decoded, "valid": True},
        "Token válido",
    )


